### DISCUSSION TITLE
ICPC2022-济南赛区 SZTU_BOOM队总结

### DISCUSSION THREAD

**Discussion by Owaiter** [CSGrandeur/s-1problem1day1ac#528] [Summary]
### 插曲

好像我们队每次正式赛开打前，志愿者都会遇到点小问题，上场沈阳是电脑打印不了换了个电脑，这场是文件解压不了卡了一会。

不过这次是小问题，开赛前还是拿到了纸质题目。

### 正赛

开场后还是我们平时的节奏，三个人随机看题，主机位的我边看题边刷新题目列表。

在第5分钟的时候，成功地刷出了本场的签到题，只不过这场有两道。我叫两个队友去看其中的M题，我自己看K题。在快速看懂K题题意后，我给队友解释了一番，讨论了一下，觉得就是个简单的STL问题，在第10分钟快速切掉了。此时正式队排名34。

K题过了之后，三个人一起看M题，好像也是我先看懂了题意，解释了一下，他们觉得交换顺序不会影响结果，我在半信半疑中把代码写了出来，交上去WA掉了。这个时候我有点怀疑是不是顺序的问题，但两个队友都觉得不影响，我重新看了一下代码，想了一下，好像最后遍历完之后还得对进位变量单独处理，跟诚钊讨论了一下，确认了是这个问题之后，改了一下，在第32分钟过了M题。这个时候其实我们罚时已经有点烂了，正式队排名210，刚好铜牌最后一名。

M题过了之后，E题过了不少人，我们立马就去看E题了。看懂了题意后，我们都觉得是个找规律题，三个人在那里狂推。我推出了一部分性质，想着有没有可能就只有这点性质，写了6个判断，偷鸡地交了一发，WA了。重新推了一下，推出了个感觉比较正确的普遍性规律，跟诚钊交流了一波，照康自己在那继续按他原来的想法推。我和诚钊觉得很有道理之后，我重新交了一发，还是WA了。这两发WA当时给我们整麻了，诚钊叫我别看这个了，别等下跟沈阳一样在一个题上吊死了，看看其他题，然后我就去看了一下全场过题第4多的A题。

A题其实我在开赛的时候就看过题意了，500的数据量，我一开始还跟他们说可能是签到题，但因为K题、M题大家过得快，所以一直都没怎么去想它。500的数据量确实很小，但我写好头文件、输入啥的之后，却无从下手，数据量太小到不知道该怎么暴力合适，在机位上发呆了好一会，诚钊和照康在旁边聊E题聊得水深火热。也不知道发呆了多久，诚钊把我赶下机了，叫我别理E题，他肯定能过，然后他就和照康边讨论边写代码，我换了个位置看A题，还是不知道从何下手，偶尔看看他们写的进度怎么样。

第147分钟，他们结束讨论，代码很快写完，测了几个样例，交上去A了。此时正式队排名288。看了一下榜单，发现铜牌区还基本只是3题，还有机会，我们三没想那么多一起看A题了。

第3小时，讨论无果，我看了一下榜单，C题、D题也是过了不少人的。C题开赛的时候我也看过了，感觉是个dp题，不擅长，就去看D题了，他们俩继续讨论A题。D题很长的题意，很明显是大模拟，我跟诚钊、照康说了一下，让他们先看D题，可能是个突破口。或许是英语六级的加持吧，我看得很快，弄懂后简单地跟他们捋了一下题意，然后我就上机敲了，叫他们还是从头开始看懂D题，生怕出现沈阳的情况（只有我一个人看完了题目，而且还理解错了题意....）。不过当我上机敲了一会，他们俩看着觉得D题太臭太长了，两个人又讨论起了A题....

敲完了输入，在我捋思路的时候，他们俩讨论出了A题一个暴力的方法，看着我没写，把我赶下机了，两个人跟E题一样，边讨论边写代码，我在旁边继续想怎么实现比较简单。

终于，在3小时半左右，他们把代码写完了，测了很多样例，感觉没错，交了上去，喜提WA。思路他俩都觉得没啥问题，继续测样例，样例也都过了。重新看了一下代码，觉得可能是边界问题，改了一下，交上去还是WA了。这个时候我们队跟E题连WA两发之后一样，有点麻了，A题样例还怎么测都是对的。

快4小时的时候，我把他俩赶下机了，看了一下榜单，已经很烂了，没有那么多顾虑了。按原来捋清的思路，我开敲D题，代码很快写完了，测了一下样例，有些地方输出了0，但很快就找到了小bug的位置，改了一下，样例都能通过。

很紧张，两个队友都没看这题，给不了我测试样例，我很怕跟沈阳一样，看错题目意思，写了个错误的代码。

复制完代码，点了一下提交按钮，在经历了20多秒焦急的等待后，“答案正确”！我舒了一口气。此时为第257分钟，粗略看了一下榜，我们的罚时大概在封榜前180名左右，还有机会，还有40分钟，我立马下机，给他俩敲A题。

此时我们也不顾及罚时了，只要过了A题，怎么样都有铜了。40分钟里，我们样例怎么测都是对的，代码怎么交都是错的，第299分钟，在交了第13发之后，比赛结束了。

### 赛后

结束后，根据前两场的情况，我们也没报什么希望了，拍了几张合照，几个人一起去湖畔吃饭。

6点多，dls终于讲完了题解，直播开始了滚榜。从封榜前第一名滚到最后一名，看着是多么漫长。到最后一名后，又从最后一名开始往前面快速地一个队一个队地解榜。当时吃饭的时候看着一个一个队过去，最后蓝色的线在我们队停下，我甚至都没反应过来，邓铭、泽驰他们就卧槽了起来，说我们最后一名。我们队三个人不愧是一队的，都没看到，跟着他们一起站起来卧槽了起来。

激动之余，因为不是亲眼所见，所以我在去做核酸的路上，还是很不敢相信。一切都太魔幻了，魔幻到我觉得是他们看错了，路上一直在问他们没看错吧。

回到实验室，我们也逐渐冷静了下来，想了下刚刚在湖畔还挺不好的，别人安静地吃饭，我们突然就站起来一直在喊卧槽。不过喊都喊了，我们打趣到可以叫老大去跟食堂要监控，明年的宣传片有素材了。

### 总结

赛前3铁了，想过很多次自己如果退役站还打铁了会怎么样。

不过一切就是这么魔幻，铜牌最后一名，解榜后看了一下，跟后一名就差了3分钟。

一直觉得自己大学跟3这个数字有缘分，没想到，这次也只是赢了3分钟....

这阵子睡觉都要笑醒了！~

### 结尾附上最终榜单

![ICPC2022济南](https://user-images.githubusercontent.com/75254346/204278251-79c47cbb-3999-4f0b-8980-acbaa4241d64.png)

**Comment by Pannta** [comment]
最后两个小时，伟凯让我看一下那道模拟题，我看了看长达三页纸的题目，回想起之前好几次比赛都被模拟题卡了几个小时，觉得这次过模拟题的希望也不大，相反，当时觉得 A 题很快能出，所以我果断放弃，回头和董哥继续研究 A 题，让伟凯一个人先研究那模拟题，结果没想到他的模拟题居然一发过了，WA 一发甚至慢一点点交就错失铜牌了，真的是关键输出。

## 凯神，我的超人！！

**Comment by dgsvygd** [comment]
## 恭喜恭喜！！！！

**Comment by CSGrandeur** [comment]
你们可以找食堂保安试试的
